File: result/translate/transcoder/python2java/java/NUMBER_VISIBLE_BOXES_PUTTING_ONE_INSIDE_ANOTHER.java
public static int f_gold ( int [ ] arr , int n ) {
  Queue < Integer > q = new LinkedList < Integer > ( ) ;
  arr = arr . clone ( ) ;
  arr . sort ( ) ;
  q . add ( arr [ 0 ] ) ;
  for ( int i = 1 ;
  i < n ;
  i ++ ) {
    int now = q . poll ( ) ;
    if ( ( arr [ i ] >= 2 * now ) && ( arr [ i ] <= 2 * now ) ) q . remove ( ) ;
    q . add ( arr [ i ] ) ;
  }
  return q . size ( ) ;
}

File: result/translate/transcoder/java2cpp/c++/NUMBER_VISIBLE_BOXES_PUTTING_ONE_INSIDE_ANOTHER.cpp
int f_gold ( int * arr , int n ) {
  queue < int > q ;
  arr = std :: copy ( arr , arr + n , queue < int > ( ) ) ;
  std :: sort ( arr , arr + n , queue < int > ( ) ) ;
  q . push ( arr [ 0 ] ) ;
  for ( int i = 1 ;
  i < n ;
  i ++ ) {
    int now = q . front ( ) ;
    if ( arr [ i ] >= 2 * now ) {
      q . pop ( ) ;
    }
    q . push ( arr [ i ] ) ;
  }
  return q . size ( ) ;
}